mailpoet/lib/Statistics/Track/SubscriberActivityTracker.php

```diff
@@ -0,0 +1,80 @@
+<?php declare(strict_types=1);
+
+namespace MailPoet\Statistics\Track;
+
+use MailPoet\Entities\SubscriberEntity;
+use MailPoet\Settings\TrackingConfig;
+use MailPoet\Subscribers\SubscribersRepository;
+use MailPoet\WP\Functions as WPFunctions;
+
+class SubscriberActivityTracker {
+
+  const TRACK_INTERVAL = 60; // 1 minute
+
+  /** @var PageViewCookie */
+  private $pageViewCookie;
+
+  /** @var SubscriberCookie */
+  private $subscriberCookie;
+
+  /** @var SubscribersRepository */
+  private $subscribersRepository;
+
+  /** @var  WPFunctions */
+  private $wp;
+
+  /** @var TrackingConfig */
+  private $trackingConfig;
+
+  public function __construct(
+    PageViewCookie $pageViewCookie,
+    SubscriberCookie $subscriberCookie,
+    SubscribersRepository $subscribersRepository,
+    WPFunctions $wp,
+    TrackingConfig $trackingConfig
+  ) {
+    $this->pageViewCookie = $pageViewCookie;
+    $this->subscriberCookie = $subscriberCookie;
+    $this->subscribersRepository = $subscribersRepository;
+    $this->wp = $wp;
+    $this->trackingConfig = $trackingConfig;
+  }
+
+  public function trackActivity(): bool {
+    if (!$this->shouldTrack()) {
+      return false;
+    }
+    $subscriber = $this->getSubscriber();
+    if (!$subscriber) {
+      return false;
+    }
+    $this->processTracking($subscriber);
+    return true;
+  }
+
+  private function processTracking(SubscriberEntity $subscriber): void {
+    $this->subscribersRepository->maybeUpdateLastEngagement($subscriber);
+    $this->pageViewCookie->setPageViewTimestamp($this->wp->currentTime('timestamp'));
+  }
+
+  private function shouldTrack() {
```

Comment:
We could add a return typehint here, probably forgotten (and new methods from later commits too).